lxd/networks_utils.go

```diff
@@ -1132,3 +1195,121 @@ func networkGetState(netIf net.Interface) api.NetworkState {
+		if err != nil {
+			switch {
+			case strings.Contains(err.Error(), "File exists"):
+			default:
+				return err
+			}
+		}
+		// add Proxy ARP record
+		_, err = shared.RunCommand("ip", "-4", "neigh", "add", "proxy", address, "dev", parentDevName)
+		if err != nil {
+			return err
+		}
+	}
+
+	// handle IPv6
+	if err := networkValidAddressV6(address); err == nil {
+		// add Local Route
+		_, err := shared.RunCommand("ip", "-6", "route", "add", "local", fmt.Sprintf("%s/128", address), "dev", parentDevName)
+		if err != nil {
+			switch {
+			case strings.Contains(err.Error(), "File exists"):
+			default:
+				return err
+			}
+		}
+		// add Proxy NDP record
+		_, err = shared.RunCommand("ip", "-6", "neigh", "add", "proxy", address, "dev", parentDevName)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func networkUnConfigureIPVlanAddress(address, parentDevName string) error {
+
+	if address == "" {
+		return fmt.Errorf("Network address can not be empty for parent nic: %s", parentDevName)
+	}
+
+	// handle IPv4
+	if err := networkValidAddressV4(address); err == nil {
+		// delete Local Route
+		_, err := shared.RunCommand("ip", "-4", "route", "del", "local", fmt.Sprintf("%s/32", address), "dev", parentDevName)
+		if err != nil {
+			switch {
+			case strings.Contains(err.Error(), "No such process"):
+			case strings.Contains(err.Error(), "No such file or directory"):
+			default:
+				return err
+			}
+		}
+		// delete Proxy ARP record
+		_, err = shared.RunCommand("ip", "-4", "neigh", "del", "proxy", address, "dev", parentDevName)
+		if err != nil {
+			switch {
+			case strings.Contains(err.Error(), "No such process"):
+			case strings.Contains(err.Error(), "No such file or directory"):
+			default:
+				return err
+			}
+		}
+	}
+
+	// handle IPv6
+	if err := networkValidAddressV6(address); err == nil {
+		// delete Local Route
+		_, err := shared.RunCommand("ip", "-6", "route", "del", "local", fmt.Sprintf("%s/128", address), "dev", parentDevName)
+		if err != nil {
+			switch {
+			case strings.Contains(err.Error(), "No such process"):
+			case strings.Contains(err.Error(), "No such file or directory"):
+			default:
+				return err
+			}
+		}
+		// delete Proxy NDP record
+		_, err = shared.RunCommand("ip", "-6", "neigh", "del", "proxy", address, "dev", parentDevName)
+		if err != nil {
+			switch {
+			case strings.Contains(err.Error(), "No such process"):
+			case strings.Contains(err.Error(), "No such file or directory"):
+			default:
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func networkGenerateInterfaceName(preffix, suffix string) (string, error) {
+	// Interface name is based upon hash (32-bit FNV-1a) of supplied preffix and suffix.
+	// Function can be used to make interface creation thread-safe and predictable.
+	// Interface name length is limited by kernel, so we use 32-bit sized hash.
+	h := fnv.New32a()
+	h.Write([]byte(fmt.Sprintf("%s-%s", preffix, suffix)))
+	name := fmt.Sprintf("eth%010d", h.Sum32())
```

Comment:
This should be based on deviceNextVeth() and probably use `ipv` as the prefix so it's clear what those are.

Reply:
deviceNextVeth can generate interface name that already exist in default namespace.
it should call func `isInterfaceNameAvailable` to avoid runtime errors

and `deviceNextVeth` should be renamed to something like `deviceNexAvailabletName` with 
prefix as argument to function

I can refactor this, if it is acceptable solution

Reply:
Yeah that should be fine

Reply:
commits that should hopefully close this issue:
https://github.com/s3rj1k/lxd/commit/62da6195caa0e9d863ed26fd4412d8c3b16fa9e6
https://github.com/s3rj1k/lxd/commit/42a98a5757525b7ecacc34ac95b67b3ffad271ab

Reply:
had to refactor this again, issues with ipvlan interface creation at start, no way on knowing interface name for lxc phys config.

https://github.com/s3rj1k/lxd/commit/c3b95e22abcaeb78d90c07b6e465cca2bc21eee9

now this function is used for veth as well, generation predictable names based on input data